src/Twig/CrudExtension.php

```diff
@@ -52,6 +55,31 @@ public function renderBreadcrumbs(array $options): string
         return '';
     }
 
+    public function generateInitialAvatar(): ?StreamInterface
+    {
+        if ($user = $this->security->getUser()) {
+            $emailParts = explode('@', $user->getUserIdentifier());
+            $name = $emailParts[0];
+            $nameParts = preg_split('/[^a-zA-Z0-9]+/', $name);
+
+            $firstName = $nameParts[0] ?? null;
+            $lastName = $nameParts[1] ?? null;
+
+            if ($firstName && $lastName) {
+                $name = sprintf('%s %s', $firstName, $lastName);
+            }
+
+            $avatar = new InitialAvatar();
+            return $avatar
+                ->name($name)
+                ->autoFont()
+                ->autoColor()
+                ->generate()->stream('data-url');
+        }
+
+        return null;
```

Comment:
never generate a non existing image! in the case there is no login. generate a generic grey `?` as user icon.

Reply:
No, in this case the src of the picture is null and the picture won't be displayed. See: https://github.com/araise-dev/CrudBundle/blob/ca0280b9f810bf510b08c95ed4527b7e8dd2ead2/src/Resources/views/includes/sidebar/_profile.html.twig#L41